### PR TITLE
CPM-594: Update Product creation modal

### PR DESCRIPTION
**Pull request by pierallard**
This PR:
- Removes the validation saying identifier should be filled for a product
- Updates the former API endpoints to have a dedicated message when user try to remove identifier / create a product without identifier ; the message links to the new endpoints usage
- Updates the new API endpoints to create products without identifiers
- Updates the filename of a PDF when it has no identifier (uuid-date.pdf instead of identifier-date.pdf)
- Adds a helper in the create product modal
- Puts family first in the create product modal, then display the identifier field if it's part of the family attributes
- Displays the UUID in the PEF breadcrumbs if product has no identifier

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Bundle/Controller/ExternalApi/ProductController.php [hunk 1]
```diff
@@ -76,6 +76,8 @@
  */
 class ProductController
 {
+    const NO_IDENTIFIER_MESSAGE = 'Validation failed. The identifier field is required for this endpoint. If you want to manipulate products without identifiers, please use products-uuid endpoints.';
```
Comment (mmetayer) [inline src/Akeneo/Pim/Enrichment/Bundle/Controller/ExternalApi/ProductController.php:79]:
```suggestion
    private const NO_IDENTIFIER_MESSAGE = 'Validation failed. The identifier field is required for this endpoint. If you want to manipulate products without identifiers, please use products-uuid endpoints.';
```

**Review by Beninho — APPROVED** [review]

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/product/form/creation/identifier-regarding-family.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+/**
+ * This compoonent allows to display an identifier field.
```
Comment (antoinearrive) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/product/form/creation/identifier-regarding-family.js:4]:
```suggestion
 * This component allows to display an identifier field.
```

File: src/Akeneo/Pim/Enrichment/Bundle/Controller/InternalApi/ProductPdfController.php [hunk 3]
```diff
@@ -72,7 +72,7 @@ public function downloadPdfAction(Request $request, string $uuid)
                 'content-type'        => 'application/pdf',
                 'content-disposition' => sprintf(
                     'attachment; filename=%s-%s.pdf',
-                    $product->getIdentifier(),
+                    $product->getIdentifier() ?? $product->getUuid(),
```
Comment (antoinearrive) [inline src/Akeneo/Pim/Enrichment/Bundle/Controller/InternalApi/ProductPdfController.php:75]: no need to use $product->getUuid()->toString()?

**Review by antoinearrive — APPROVED** [review]

File: src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php [hunk 4]
```diff
@@ -175,14 +175,6 @@ public function it_updates_a_product_s_identifier(): void
         Assert::assertSame('new_identifier', $updatedProduct->getValue('sku')?->getData());
     }
 
-    /** @test */
-    public function it_throws_an_exception_when_deleting_the_identifier_value()
```
Comment (mmetayer) [inline src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php:179]: you could replace this test by one checking the identifier value was successfully deleted

File: src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php [hunk 5]
```diff
@@ -1435,21 +1427,6 @@ public function it_creates_a_product_without_identifier_nor_uuid_as_param()
         Assert::assertSame('product_sku', $product->getIdentifier());
     }
 
-    /**
-     * @test
-     * @TODO: remove this test once the identifier is rendered optional
-     */
-    public function it_cannot_create_a_product_without_identifier_value(): void
```
Comment (mmetayer) [inline src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php:1442]: same, test that we can create a product without identifier now

**Review by cemorin — APPROVED** [review]

**Review by mmetayer — APPROVED** [review]

**Pull request merged**